From f6288f8660d4683878844142862f9a915dccff39 Mon Sep 17 00:00:00 2001
From: Suchir Kavi <suchirkavi@gmail.com>
Date: Mon, 26 May 2025 17:42:02 -0700
Subject: [PATCH 1/2] Revert "libfs: Fix duplicate directory entry in
 offset_dir_lookup"

This reverts commit 176d0333aae43bd0b6d116b1ff4b91e9a15f88ef.
---
 fs/libfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/libfs.c b/fs/libfs.c
index 3cb49463a849..b0f262223b53 100644
--- a/fs/libfs.c
+++ b/fs/libfs.c
@@ -492,7 +492,7 @@ offset_dir_lookup(struct dentry *parent, loff_t offset)
 		found = find_positive_dentry(parent, NULL, false);
 	else {
 		rcu_read_lock();
-		child = mas_find_rev(&mas, DIR_OFFSET_MIN);
+		child = mas_find(&mas, DIR_OFFSET_MAX);
 		found = find_positive_dentry(parent, child, false);
 		rcu_read_unlock();
 	}
-- 
2.49.0

